src/kernels/MIOpenBatchNormBwdSpatial.cl

```diff
@@ -429,7 +429,7 @@ MIOpenBatchNormBwdSpatial(const __global _FLOAT* __restrict x_in,
     nidx                = remkey / MIO_BN_HW;
     hwidx               = remkey - (nidx * MIO_BN_HW);
     index               = nidx * MIO_BN_CHW + chwid + hwidx;
-    if(index < MIO_BN_NCHW)
+    if(index < (MIO_BN_NCHW - 3))
```

Comment:
It is possible that MIO_BN_NCHW <= 3 here?

Reply:
We are reading 4 inputs at a time.  Here index is going beyond MIO_BN_NCHW and hence I put it to (MIO_BN_NCHW -3 ) to cap it at  MIO_BN_NCHW.

Reply:
We are reading 4 inputs at a time and indexing is going beyond MIO_BN_NCHW.  I am limiting at (MIO_BN_NCHW -3) to cap it at MIO_BN_NCHW.

Reply:
The answer is "No"?

Reply:
yes Artem